plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/N4JSProjectSetupJsonValidatorExtension.xtend

```diff
@@ -1166,8 +1166,13 @@ public class N4JSProjectSetupJsonValidatorExtension extends AbstractPackageJSONV
 				return;
 			}
 
-			val currNodeModulesFolder = allNodeModuleFolders.get(currentProjectName)?.toPath;
-			val currNPM = currNodeModulesFolder?.resolve(id.rawName);
+			var Path currNPM = null;
+			for (File nodeModulesFolder : allNodeModuleFolders.get(currentProjectName)) {
+				if (currNPM === null || !currNPM.toFile.exists) {
+					currNPM = nodeModulesFolder.toPath.resolve(id.rawName);
+				}
```

Comment:
This looks for the npm package `id` in both the local and global node_modules folders. It seems this check should be performed in a well-defined order, to always obtain the same result in case the project exists in both the local and global node_modules folders. Even if we used a LinkedHashMap (which we don't, I think?) it seems to me it would be good to make this more explicit.